src/chunkserver/scan_manager.cpp

```diff
@@ -21,156 +21,300 @@
+                    job->type = ScanType::NewMap;
+                    job->iter++;
+                    job->currentChunkId = job->iter->first;
                 }
                 break;
             case ScanType::Finish:
-                ScanJobFinish();
+                ScanJobFinish(key);
                 done = true;
                 break;
             default:
+                LOG(WARNING) << "Error scan type.";
                 break;
         }
     }
 }
 
-ChunkID ScanManager::GetNextScanChunk() {
-    return 1;
+// send scan request to braft
+void ScanManager::ScanChunkReqProcess(const std::shared_ptr<ScanJob> job) {
+    // split scan chunk request
+    uint32_t currentOffset = 0;
+    auto chunkSize = copysetNodeManager_->GetCopysetNodeOptions().maxChunkSize;
+    auto nodePtr = copysetNodeManager_->GetCopysetNode(job->poolId, job->id);
+    while (currentOffset < chunkSize) {
+        ChunkResponse *response = new ChunkResponse();
+        ChunkRequest request;
+        request.set_optype(CHUNK_OP_TYPE::CHUNK_OP_SCAN);
+        request.set_logicpoolid(job->poolId);
+        request.set_copysetid(job->id);
+        request.set_chunkid(job->currentChunkId);
+        request.set_offset(currentOffset);
+        request.set_size(scanSize_);
+        // reuse the clonefileoffset field to deliver rpc's timeout
+        request.set_clonefileoffset(timeout_);
+
+        std::shared_ptr<ScanChunkRequest> req =
+            std::make_shared<ScanChunkRequest>(nodePtr, this, &request,
+                                               response, nullptr);
+        req->Process();
+        currentOffset += scanSize_;
+    }
 }
 
-void ScanManager::ScanError() {
-    return;
+void ScanManager::SetLocalScanMap(ScanKey key, const ScanMap &map) {
+    auto job = GetJob(key);
+    if (nullptr != job) {
+        job->localMap = map;
+        job->waitingNum--;
+    }
 }
 
-void ScanManager::ScanChunkReqProcess() {
-    return;
+void ScanManager::DealFollowerScanMap(const FollowScanMapRequest &request,
+                         FollowScanMapResponse *response) {
+    ScanMap scanMap = request.scanmap();
+    ScanKey key(scanMap.logicalpoolid(), scanMap.copysetid());
+    auto job = GetJob(key);
+    if (nullptr != job) {
+        job->repMap.emplace_back(scanMap);
+        job->waitingNum--;
+        job->type = ScanType::WaitMap;
+        GenScanJob(key);
+        response->set_retcode(CHUNK_OP_STATUS::CHUNK_OP_STATUS_SUCCESS);
+    } else {
+        response->set_retcode(CHUNK_OP_STATUS::CHUNK_OP_STATUS_INVALID_REQUEST);
+    }
 }
 
-void ScanManager::BuildLocalScanMap() {
-    return;
+void ScanManager::CompareMap(ScanKey key) {
+    auto job = GetJob(key);
+    if (nullptr != job) {
+        // check there are three scanmap
+        if (job->repMap.size() != 2 || job->localMap.logicalpoolid() == 0) {
+            LOG(ERROR) << "Compare scanmap failed,"
+                       << " there are not three scanmaps."
```

Comment:
there are not three scanmaps -> The information of the replication group is not collected completely？

Reply:
> there are not three scanmaps -> The information of the replication group is not collected completely？

Yes, but find a bug in design, will improve later.